web/playground/src/sidebar/Sidebar.js

```diff
@@ -1,24 +1,58 @@
 import "./Sidebar.css";
 import React from "react";
+import { useState } from "react";
 
 function Sidebar({ library, onLoadFile }) {
   function loadFile(section, file) {
     onLoadFile(file, library[section][file]);
   }
 
+  function toggleFolder(id) {
+    openFolders[id] = !Boolean(openFolders[id]);
+    setOpenFolders(() => openFolders);
+  }
+
+  function handleClick(section, file, id) {
+    if (isFile(file)) {
+      loadFile(section, file);
+    } else {
+      toggleFolder(id);
+    }
+    sv(($) => $ + 1);
+  }
+
+  function isFile(path) {
+    return path.endsWith(".prql");
+  }
+
   const sections = [];
+  const [openFolders, setOpenFolders] = useState({});
+  const [v, sv] = useState(1);
 
   for (const [section, files] of Object.entries(library)) {
     const fileRows = [];
     for (const [index, filename] of Object.keys(files).entries()) {
+      const array = files[filename];
+      const depth = array[2];
+      const parent = array[3];
+      const id = array[4];
+      const name = array[5];
```

Comment:
At some point, this will have to be refactored from arrays into objects. I knew it when I changed `source` into `[output, source]`.

Reply:
Sure. This will need changes for the examples and chinook as well

Reply:
And it may not be compatble with currently saved queries in local storage

Reply:
Oh yeah, I haven't though about that. Well, we'll do it some time later then.